File: ANTWebAPI/src/ANTWebAPI.PL/DTOs/ArticleDTO.cs
namespace ANTWebAPI.PL.DTOs;

public class ArticleDTO
{
    public long Id { get; init; }
    public CatalogDTO Catalog { get; init; } = null!;
    public string Title { get; init; } = string.Empty;
    public string Description { get; init; } = string.Empty;
    public string DateOrBanner { get; init; } = string.Empty;
}

File: ANT-Android-App/data/src/main/java/com/fredprojects/ant/data/remote/dto/ArticleDto.kt
package com.fredprojects.ant.data.remote.dto

import kotlinx.serialization.Serializable

/**
 *  The data transfer object for articles received from the server
 *  @param id is a unique id of the article
 *  @param catalog is a type of the article (see [CatalogDto])
 *  @param title is a title of the article
 *  @param description is a description of the article
 *  @param dateOrBanner is a date or a banner of the article
 *  @param content is a list of strings representing the content of the article
 */
@Serializable
data class ArticleDto(
    val id: Long,
    val catalog: CatalogDto,
    val title: String,
    val description: String,
    val dateOrBanner: String,
    val content: List<String>
)